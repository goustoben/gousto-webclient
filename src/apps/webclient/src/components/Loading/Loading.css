.gif {
	width: 150px;
	height: auto;
}

.tastePreferences {
	width: 100px;
	height: auto;
	margin: 0 auto;
	display: block;
}

.tastePreferencesContent {
	text-align: center;
	padding: 1.5rem 0;
}
