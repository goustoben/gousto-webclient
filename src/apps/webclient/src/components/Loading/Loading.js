import React from 'react'
import PropTypes from 'prop-types'
import classNames from 'classnames'
import { Div } from 'Page/Elements'
import Image from 'Image'

import css from './Loading.css'

const getImage = (fileName) => require(`media/images/${fileName}`) // eslint-disable-line global-require

export const LoadingOverlay = () => (
  <Div
    fixed
    w-100
    style={{ zIndex: 100, top: 0, left: 0 }}
  >
    <Div
      d-block
      margin={{
        top: 'XXL',
        right: 'auto',
        left: 'auto',
      }}
      style={{
        width: '100px',
        pointerEvents: 'none',
      }}
    >
      <Loading />
    </Div>
  </Div>
)

const Loading = ({ className }) => (
  <Image media={getImage('Loading-Icon.gif')} className={classNames(css.gif, className)} />
)

Loading.propTypes = {
  className: PropTypes.string,
}

Loading.defaultProps = {
  className: null,
}

export default Loading
