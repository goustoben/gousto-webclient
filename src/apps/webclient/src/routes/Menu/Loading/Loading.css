.container {
	position: fixed;
	z-index: 1;
	width: 100%;
}

.loading {
	display: block;
	margin: auto;
	width: 100px;
	margin-top: 60px;
	pointer-events: none;
}

.loading--recommendations {
	display: block;
	width: 160px;
	margin: 100px auto 0;
	text-align: center;
}

.spinner {
	height: 150px;
	width: 150px;
}

.loading-text {
	display: inline;
	font-weight: 600;
}

	.loading-text__title {
		display: inline-block;
	}

	.loading-text__sub {
		font-size: 24px;
		font-weight: 600;
	}

.icon {
	height: 12px;
	width: 14px;
	margin: 0 1px;
}
