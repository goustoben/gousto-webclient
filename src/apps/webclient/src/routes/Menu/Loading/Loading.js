import PropTypes from 'prop-types'
import React from 'react'
import LoadingComponent from 'Loading'

import css from './Loading.css'

const propTypes = {
  loading: PropTypes.bool,
}

const defaultProps = {
  loading: false,
}

const Loading = ({ loading }) => {
  const loadingCss = css.loading

  return (
    (loading) ? (
      <div className={css.container}>
        <div className={loadingCss}>
          <LoadingComponent />
        </div>
      </div>
    ) : null
  )
}

Loading.propTypes = propTypes

Loading.defaultProps = defaultProps

export default Loading
