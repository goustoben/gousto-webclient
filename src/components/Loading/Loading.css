.gif {
	width: 150px;
	height: auto;
}
