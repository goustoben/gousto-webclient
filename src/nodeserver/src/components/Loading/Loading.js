import React from 'react'
import { Div } from 'Page/Elements'
import Image from 'Image'
import css from './Loading.css'

const getImage = (fileName) => require(`media/images/${fileName}`) // eslint-disable-line global-require

const Loading = () => (
	<Image media={getImage('Loading-Icon.gif')} title="animation" className={css.gif} />
)

export const LoadingOverlay = () => (
	<Div
		fixed
		w-100
		style={{ zIndex: 100, top: 0, left: 0 }}
	>
		<Div
			d-block
			margin={{
				top: 'XXL',
				right: 'auto',
				left: 'auto',
			}}
			style={{
				width: '100px',
				pointerEvents: 'none',
			}}
		>
			<Loading />
		</Div>
	</Div>
)

export default Loading
