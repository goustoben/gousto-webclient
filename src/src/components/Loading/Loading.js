import React from 'react'
import PropType from 'prop-types'
import { Div } from 'Page/Elements'
import Image from 'Image'

import css from './Loading.css'

const getImage = (fileName) => require(`media/images/${fileName}`) // eslint-disable-line global-require

export const LoadingTastePreferences = () => (
  <div>
    <Image
      media={getImage('Loading-Icon-building-menu.gif')}
      title="animation"
      className={css.tastePreferences}
    />
    <p className={css.tastePreferencesContent}><strong>Building your menu</strong></p>
  </div>
)

export const LoadingOverlay = () => (
  <Div
    fixed
    w-100
    style={{ zIndex: 100, top: 0, left: 0 }}
  >
    <Div
      d-block
      margin={{
        top: 'XXL',
        right: 'auto',
        left: 'auto',
      }}
      style={{
        width: '100px',
        pointerEvents: 'none',
      }}
    >
      <Loading />
    </Div>
  </Div>
)

const Loading = ({ className }) => (
  <Image media={getImage('Loading-Icon.gif')} title="animation" className={`${css.gif} ${className}`} />
)

Loading.propTypes = {
  className: PropType.string,
}

export default Loading
