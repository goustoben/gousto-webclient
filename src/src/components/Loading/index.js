export { default as Component } from './Loading'
export * from './Loading'
