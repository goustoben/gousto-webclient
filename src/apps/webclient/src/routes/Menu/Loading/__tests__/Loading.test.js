import React from 'react'
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer'
import Loading from '../Loading'

describe('Menu Loading', () => {
  let wrapper

  describe('Initial Rendering', () => {
    test('should match snapshot', () => {
      const tree = renderer.create(<Loading loading hasRecommendations />).toJSON()
      expect(tree).toMatchSnapshot()
    })
  })

  describe('Alternative Rendering', () => {
    test('should not render anything if loading is null', () => {
      wrapper = shallow(<Loading />)

      expect(wrapper.type()).toBe(null)
    })

    test('should render loading component if hasRecommendations is false', () => {
      wrapper = shallow(<Loading loading />)
      const loadingComponent = wrapper.find('[className="loading"]')

      expect(loadingComponent).toHaveLength(1)
    })
  })
})
