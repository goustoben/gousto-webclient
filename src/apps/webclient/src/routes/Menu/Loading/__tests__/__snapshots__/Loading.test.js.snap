// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu Loading Initial Rendering should match snapshot 1`] = `
<div
  className="container"
>
  <div
    className="loading"
  >
    <img
      alt=""
      className="gif contain"
      onClick={[Function]}
      src="Loading-Icon.gif"
    />
  </div>
</div>
`;
